### PR TITLE
Overall bug fixes

### PR DESCRIPTION
**Pull request by yshdb5**
### Description:
This pull request addresses various bugs identified during UI testing, refines the overall user experience, and introduces functional and display improvements. Below is a comprehensive list of resolved issues, feature updates, and specific changes implemented to enhance UI consistency and functionality across request and provider screens.

- **What issue does it solve?** This PR resolves issues including a bug that removed images when editing requests, readability of service types and statuses, and an unnecessary `BottomNavigationMenu` in the `CreateRequestScreen`. It also fixes a dark mode bug that hid certain requests and prevents a random crash in `ProviderProfile.kt`.
- **What functionality does it add?** This PR adds standardized formatting across request screens, a refined `SelectProviderScreen` with filtered providers by service type, and updates to the `RequestsOverviewScreen` for display consistency.

### Related Issues:
- #304 

### Changes Made:
- **UI Improvements:**
  - Removed `BottomNavigationMenu` from `CreateRequestScreen` to streamline the user interface.
  - Improved layout of `ShortcutsSection` components for better display adaptability to screen size and font changes.
  - Updated `RequestsOverviewScreen` with `Services.format()` and `ServiceRequestStatus.format()` methods to ensure consistent display of service types and statuses.
  - Enhanced `SelectProviderScreen` with a default profile picture, adjusted description text size, and refined rating display.
  
- **Functionality Enhancements:**
  - Added filtering of providers by service type in `SelectProvider.kt`.
  - Added "Not implemented" toast notifications to certain buttons.
  - Updated default header picture for `SelectProviderScreen`.
  
- **Bug Fixes:**
  - Resolved an image deletion bug in `EditRequestScreen` during request editing.
  - Fixed dark mode issue that hid requests on certain screens.
  - Prevented random crash in `ProviderProfile.kt`.

### How to Test:
1. Go to `CreateRequestScreen` and confirm the absence of the `BottomNavigationMenu`.
2. Edit a request in `EditRequestScreen` with an associated image, save without modifying the image, and verify that it remains intact.
3. In `RequestsOverviewScreen`, confirm consistent formatting for service types and request statuses.
4. On `SelectProviderScreen`, verify the default profile picture, description text size, and rating display.
5. Test `SelectProviderScreen` to ensure providers are filtered correctly by service type.
6. Test that overall buttons work or display "Not implemented" toast notifications.

---

### Checklist:
- [x] Code adheres to project coding standards.
- [x] New and updated tests have been written and verified.
- [x] Functionality has been manually tested.
- [x] Code is free of known issues.
- [x] Documentation has been updated where applicable.

### CODE REVIEW

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt:1]: Good improvements to the code to correspond with the actual behavior we want.

**Review by achillet3 — COMMENTED** [review]
# Overall Comments
- Very good changes addressing the large majority of the bugs we noted throughout the app
- The UI is more intuitive and has fewer breaks. The app is starting to look seriously good!

# Bug found
- When creating a request, the location, when filled, returns an invalid field error. You don't seem to modify the code, so it's probably from a previousin, merge to the ma but it could be great to fix.

**Comment by yshdb5** [comment]
> # Bug found
> * When creating a request, the location, when filled, returns an invalid field error. You don't seem to modify the code, so it's probably from a previousin, merge to the ma but it could be great to fix.

Thank you for pointing it out! Good catch, it was indeed modified by somebody in a previous PR. It is now fixed. I have also pushed additional improvements to the overall UI. Let me know if everything checks out!

**Review by achillet3 — APPROVED** [review]
Thanks for the changes. LGTM!

**Review by ethanboren — CHANGES_REQUESTED** [review]
Thank you for this high-quality contribution. The pull request is very comprehensive and addresses several important bugs and improvements. Overall, the work is very well done and significantly enhances the user experience.

However, I have some additional points to mention:
1. **Issue with accented characters in `CreateRequestScreen`:** Currently, it is impossible to input accented characters in `CreateRequestScreen`. This is important for users who need to enter text in languages that use accents. Please address this bug to improve the user experience.

2. **Selecting past dates in CreateRequestScreen:** The application allows users to select a date that has already passed when creating a request in `CreateRequestScreen`. It would be better to prevent the selection of past dates to avoid inconsistencies and potential errors.

3. **Border colors and field validation in `CreateRequestScreen`:** It would be beneficial to add colors to the borders of input fields to improve visibility and ergonomics. Similar to when creating an account as a seeker or provider, fields should turn red if the information is incorrectly filled, accompanied by an error message. Currently, it is difficult for the user to know which fields are mandatory because when pressing the submit button, no toast or error message appears to explain the problem.

4. **Button without action in `RequestsOverviewScreen`:** In `RequestsOverviewScreen`, the button with the three horizontal lines at the top right does nothing when clicked, and there is no toast message indicating that the feature is not implemented. It would be appropriate to add a toast message to inform the user that this functionality is forthcoming.

5. **Text loss when switching modes in `CreateRequestScreen`:** When entering text in "Create a new request" for the seeker and switching modes (from light mode to dark mode or vice versa), the entered text disappears, and you have to start over. This can be frustrating for the user. It is important to preserve the text state when changing modes.


Other small point to pay attention to :

1. Consistency of Toast messages: For the "Not implemented" notifications, please ensure that the message is consistent across all views where it is used.

Once again, excellent work on this pull request. After these small adjustments, it will be ready to be merged.

**Comment by yshdb5** [comment]
> 1. **Issue with accented characters in `CreateRequestScreen`:** Currently, it is impossible to input accented characters in `CreateRequestScreen`. This is important for users who need to enter text in languages that use accents. Please address this bug to improve the user experience.

Thanks! I do not know why this happens, OutlinedTextFields are supposed to handle every utf-8 characters. Please tell me if you know how to fix this and I'll implement it in a later PR.

> 2. **Selecting past dates in CreateRequestScreen:** The application allows users to select a date that has already passed when creating a request in `CreateRequestScreen`. It would be better to prevent the selection of past dates to avoid inconsistencies and potential errors.

You are right, after some research, I do not see an easy fix for this problem right now. Maybe we can implement it in a future sprint and hope that for now, seekers won't create service requests for a past dates...

> 3. **Border colors and field validation in `CreateRequestScreen`:** It would be beneficial to add colors to the borders of input fields to improve visibility and ergonomics. Similar to when creating an account as a seeker or provider, fields should turn red if the information is incorrectly filled, accompanied by an error message. Currently, it is difficult for the user to know which fields are mandatory because when pressing the submit button, no toast or error message appears to explain the problem.

The creation and edition of a ServiceRequest is less sensitive than entering credentials and/or passwords. I think that the actual checks on the date and location together with the fact that it is not possible to submit a request without a title, description and service type is sufficient for now.

> 4. **Button without action in `RequestsOverviewScreen`:** In `RequestsOverviewScreen`, the button with the three horizontal lines at the top right does nothing when clicked, and there is no toast message indicating that the feature is not implemented. It would be appropriate to add a toast message to inform the user that this functionality is forthcoming.

Fixed, thanks!

**Comment by ethanboren** [comment]
Thank you for your response.

1. For the first point, I don't have much of an idea either about why this is happening; it seems strange to me as well. I'll look into it more deeply and get back to you with an answer next week.
2. For the second and third points, keep in mind that they need to be modified by next week because these are bugs that must be fixed.
3. For the fourth point, thank you for fixing it!

Good work in general on this PR.

**Pull request merged**